core/meta/src/TStreamerElement.cxx

```diff
@@ -305,8 +306,8 @@ TClass *TStreamerElement::GetClassPointer() const
 
 Int_t TStreamerElement::GetExecID() const
 {
-   //check if element is a TRef or TRefArray
-   if (strncmp(fTypeName.Data(),"TRef",4) != 0) return 0;
+   TClass* const type = GetClassPointer();
+   if (!type || !(type->InheritsFrom(TRef::Class()) || type->InheritsFrom(TRefArray::Class()) || type->InheritsFrom("TRefProxy") || type->InheritsFrom("TRefArrayProxy"))) return 0;
```

Comment:
If the last 2 are needed, you can use just `type->InheritsFrom("TVirtualRefProxy")`